packages/plugin-react-native-client-sync/client-sync.js

```diff
@@ -98,6 +98,16 @@ module.exports = (NativeClient) => ({
         case 'ContextUpdate':
           origSetContext.call(client, event.data)
           break
+        case 'AddFeatureFlag':
+          origAddFeatureFlag.call(client, event.data.name, event.data.variant)
+          break
+        case 'ClearFeatureFlag':
+          if (event.data && event.data.name) {
+            origClearFeatureFlag.call(client, event.data.name)
+          } else {
+            origClearFeatureFlags.call(client)
+          }
+          break
```

Comment:
Why are feature flags being synced with distinct add and clear events rather than following the approach used for metadata where a single 'update' event is used to update everything at once?

Reply:
This was mostly because unlike metadata, feature flags are "write only". So while they can be changed, there is no API-level way to read them back. Adding an internal-only read API on the native notifiers would solve this though (and probably make a few other things simpler at the same time).

Reply:
Thanks for the explanation!